src/NServiceBus.RabbitMQ/Routing/DefaultRoutingKeyConvention.cs

```diff
@@ -23,6 +24,16 @@ static string GetRoutingKey(Type type, string key = "")
 
             var implementedInterface = interfaces.FirstOrDefault();
 
+            if (interfaces.Count > 0)
+            {
+                Logger.Warn("The direct routing topology might not deliver message types with interfaces in their inheritance hierarchy. Message types used for publishing should not implement interfaces.");
```

Comment:
@adamralph I suppose that scenario could happen, but can you come up with a way to accurately and succinctly explain which interfaces are the "bad" ones here?  Both "non-system" and "custom" don't really mean anything without an explanation of what we mean by those terms.